### PR TITLE
[JENKINS-70906] Compatibility with Prototype removal

### PR DESCRIPTION
**Pull request by daniel-beck**
Based on https://github.com/jenkinsci/matrix-auth-plugin/pull/139, actual changes are from 4th commit onward.

Compatibility with [JENKINS-70906](https://issues.jenkins.io/browse/JENKINS-70906).

Tested with and without https://github.com/jenkinsci/jenkins/pull/7986, including ambiguity-related code and SystemRead / ExtendedRead.

### CODE REVIEW

File: src/main/resources/hudson/security/table.js [hunk 1]
```diff
@@ -40,15 +40,15 @@ Behaviour.specify(".matrix-auth-add-button", 'GlobalMatrixAuthorizationStrategy'
 
     var tooltipAttributeName = getTooltipAttributeName();
 
-    findElementsBySelector(copy, ".stop a").each(function(item) {
+    findElementsBySelector(copy, ".stop a").forEach(function(item) {
```
Comment (daniel-beck) [inline src/main/resources/hudson/security/table.js:43]: Tooltips for actions in newly added rows don't have placeholders, but proper labels.

File: src/main/resources/hudson/security/table.js [hunk 2]
```diff
@@ -40,15 +40,15 @@ Behaviour.specify(".matrix-auth-add-button", 'GlobalMatrixAuthorizationStrategy'
...
       let oldTitle = item.getAttribute("title");
       if (oldTitle !== null) {
         item.setAttribute("title", oldTitle.replace("__SID__", name).replace("__TYPE__", typeLabel));
       }
       item.setAttribute(tooltipAttributeName, item.getAttribute(tooltipAttributeName).replace("__SID__", name).replace("__TYPE__", typeLabel));
     });
 
-    findElementsBySelector(copy, "input[type=checkbox]").each(function(item) {
+    findElementsBySelector(copy, "input[type=checkbox]").forEach(function(item) {
```
Comment (daniel-beck) [inline src/main/resources/hudson/security/table.js:51]: Tooltips for checkboxes in newly added rows don't have placeholders, but proper labels.

File: src/main/resources/hudson/security/table.js [hunk 3]
```diff
@@ -175,7 +175,7 @@ Behaviour.specify(".global-matrix-authorization-strategy-table TD.stop A.migrate
         if(ambiguousPermissionInputs[i].type == "checkbox") {
           unambiguousPermissionInputs[i].checked |= ambiguousPermissionInputs[i].checked;
         }
-        newNameElement.className += ' highlight-entry';
+        newNameElement.classList.add(' highlight-entry');
```
Comment (daniel-beck) [inline src/main/resources/hudson/security/table.js]: Merge ambiguous row into existing unambiguous row (typically authenticated/anonymous)

File: src/main/resources/hudson/security/table.js [hunk 4]
```diff
@@ -137,7 +137,7 @@ Behaviour.specify(".global-matrix-authorization-strategy-table TD.stop A.migrate
     var name = tr.getAttribute('name');
 
     var newName = name.replace('[EITHER:', '[USER:'); // migrate_user behavior
-    if (this.hasClassName('migrate_group')) {
+    if (this.classList.contains('migrate_group')) {
```
Comment (daniel-beck) [inline src/main/resources/hudson/security/table.js:140]: Identify different behavior for the "migrate to group" button.

File: src/main/resources/hudson/security/table.js [hunk 5]
```diff
@@ -222,7 +222,7 @@ function getTooltipAttributeName() {
  */
 Behaviour.specify(".global-matrix-authorization-strategy-table td input", 'GlobalMatrixAuthorizationStrategy', 0, function(e) {
   var table = findAncestor(e, "TABLE");
-  if (table.hasClassName('read-only')) {
+  if (table.classList.contains('read-only')) {
```
Comment (daniel-beck) [inline src/main/resources/hudson/security/table.js:225]: Identify SystemRead / ExtendedRead config forms and keep all checkboxes disabled.

**Review by basil — APPROVED** [review]

**Pull request merged**